### ISSUE TITLE
Questions about the optimization

### ISSUE THREAD

**Issue by stwerner97** [lvwerra/trl#178]
Thanks so much for your a great work!

I have a question about the number of optimization steps the reinforcement learning algorithm needs before converging. 

In more traditional optimization tasks (i.e., outside of natural language processing), reinforcement learning often need orders of magnitude more steps to find suitable strategies. Still, the optimization in [gpt2-sentiment](https://github.com/lvwerra/trl/blob/main/examples/sentiment/notebooks/gpt2-sentiment.ipynb) converges after less than a hundred gradient steps.

Could you give some intuition as to how this is possible?

**Comment by lvwerra** [comment]
I think the main reason is that language models are pretrained and in the gpt2-sentiment example further fine-tuned on movie reviews. That means the model is already good at generating movie reviews before RL tuning and thus essentially short-cutting the exploration phase of RL.

**Comment by stwerner97** [comment]
Thank you for your response!

Although I was thinking along similar lines, I am still surprised that the gradient steps start improving the results forthwith, since I believe that the agent's value estimates would need to match the pre-trained behaviour before gradient updates can start improving the generation of movie reviews. So I would have expected the results to deteriorate first.
